docs/source/docker/integration/s3fs-fuse.rst

```diff
@@ -0,0 +1,209 @@
+
+Load data directly from S3 buckets using s3fs-fuse
+===================================================
+
+Learn how to use Lightly Docker directly with an AWS S3 bucket as input.
+
+Very often ML teams don't have all the data locally on a single machine. Instead,
+the data is stored in a cloud storage such as AWS S3 or Google Cloud Storage.
+Wouldn't it be great to directly connect to these storage buckets instead of
+having to download the data everytime to your local machine?
+
+In this example we will do the following:
+
+1. Setup `s3fs-fuse <https://github.com/s3fs-fuse/s3fs-fuse>`_ to mount an S3 bucket to your local machine storage
+2. Use the Lightly docker and set the input directory to the mounted S3 bucket
+
+What is s3fs-fuse?
+--------------------
+
+`s3fs-fuse <https://github.com/s3fs-fuse/s3fs-fuse>`_ is an open-source project
+that allows to mount an S3 bucket to a local file storage. Only a limited set of
+file operations are supported (e.g. appending data to a file is very inefficient and slow 
+as there is no direct support). However, for just reading files to train a ML model
+this approach is more than sufficient.
+
+Here are some of the limitations pointed out in the GitHub readme:
+
+.. note:: Generally S3 cannot offer the same performance or semantics as a local file system. More specifically:
+  
+  - random writes or appends to files require rewriting the entire object, optimized with multi-part upload copy
+  - metadata operations such as listing directories have poor performance due to network latency
+  - non-AWS providers may have eventual consistency so reads can temporarily yield stale data (AWS offers read-after-write consistency since Dec 2020)
+  - no atomic renames of files or directories
+  - no coordination between multiple clients mounting the same bucket
+  - no hard links
+  - inotify detects only local modifications, not external ones by other clients or tools
+
+Get an AWS Bucket and credentials
+-----------------------------------
+
+From the AWS dashboard go the **S3** service (https://s3.console.aws.amazon.com/s3/home)
+and create an S3 bucket if you don't have one yet. 
+
+If you don't have credentials yet you need to go to the **IAM** service 
+(https://console.aws.amazon.com/iam/home) on AWS and create
+a new user. Make sure you add the **AmazonS3FullAccess** permission. Then create
+and download the credentials (.csv file). In the credentials file you should find
+the **Access key ID** and **Secret access key** we will use later.
+
+Install s3fs-fuse
+-------------------
+
+In order to install `s3fs <https://github.com/s3fs-fuse/s3fs-fuse>`_ we can follow the instructions from the GitHub readme.
+On `Debian 9 or newer` or `Ubuntu 16.04 or newer` we can use the following terminal instructions to install it:
+
+.. code-block:: console
+
+    sudo apt install s3fs
+
+
+Below we show the output for installing s3fs on a Google Cloud Compute instance.
+
+.. code-block:: console
```

Comment:
Is this really necessary?

Reply:
I think it's useful to see what one would expect to see.